src/liballoc/arc.rs

```diff
@@ -990,11 +997,7 @@ impl<T> Weak<T> {
     pub fn new() -> Weak<T> {
         unsafe {
             Weak {
-                ptr: Box::into_raw_non_null(box ArcInner {
-                    strong: atomic::AtomicUsize::new(0),
-                    weak: atomic::AtomicUsize::new(1),
-                    data: uninitialized(),
-                }),
+                ptr: NonNull::new_unchecked(WEAK_EMPTY as *mut _),
```

Comment:
That picks a value based on the alignment size of `T`. While `RawVec` can check `cap` if its pointer is actually null, `Weak` doesn't have that. So, to check, `dangling` would need to be called each time.

It's also not clear to me what having a never-used pointer to `8`, or `16`, etc would provide.

Reply:
I don't think calling `dangling` every time is a problem.

Reply:
It shouldn't be a problem, the value is constant for the input type.

It may avoid memory access errors in debuggers, or avoid tripping unaligned access warnings in tools like Valgrind (if that's a thing).

Reply:
In https://github.com/rust-lang/rust/pull/41064 we changed all dangling pointers to be aligned rather than 1. There is no reason for this case to be different.

`dangling()` optimizes to a constant, there is not issue with calling it many times.